### PR TITLE
DROOLS-7363 refactor tests

### PR DESCRIPTION
**Pull request by nprentza**
the PR introduces a new class for testing with a 2-rule drl, the basic rule plus one that updates a fact under conditions.
the PR also suggests the introduction of the abstract class ReliabilityTestBasics for the common functions the various test classes will share.

### CODE REVIEW

File: drools-reliability/src/test/java/org/drools/reliability/ReliabilityTestBasics.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    }
+
+    static Stream<PersistedSessionOption.Strategy> strategyProviderStoresOnly() {
+        return Stream.of(PersistedSessionOption.Strategy.STORES_ONLY);
+    }
+
+    @AfterEach
+    public void tearDown() {
+        // We can remove this when we implement ReliableSession.dispose() to call CacheManager.removeCachesBySessionId(id)
+        CacheManager.INSTANCE.removeAllSessionCaches();
```
Comment (mariofusco) [inline drools-reliability/src/test/java/org/drools/reliability/ReliabilityTestBasics.java:52]: I think that this method should also call `failover()` or at least reset the counter of sessions id. If you don't do so I experienced a coupling between different tests, succeeding in isolation but failing when launched the whole test class.
@tkobayas can you please provide your feedback about this?

Comment (tkobayas) [reply drools-reliability/src/test/java/org/drools/reliability/ReliabilityTestBasics.java:52]: @mariofusco Ahh, okay. The root problem is `savedSessionId`. It is changed every time `ReliabilityTest.getKieSession` is called, so even if a test starts with sessionId = 1, after `failover` + `restoreSession`, `savedSessionId` becomes 0. It broke the tests. I raised a fix PR.

https://github.com/kiegroup/drools/pull/5104

The reason why `failover` in `tearDown` solved the issue was it keeps tests start with sessionId = 0, so the problem wouldn't arise (same for single method test running).

with the PR fix, we no longer need  `failover` in `tearDown`.

File: drools-reliability/src/test/java/org/drools/reliability/ReliabilityTestBasics.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    protected KieSession restoreSession(String drl, PersistedSessionOption.Strategy strategy) {
+        return getKieSession(drl, PersistedSessionOption.fromSession(savedSessionId, strategy));
+    }
+
+    protected KieSession getKieSession(String drl, PersistedSessionOption option) {
+        KieBase kbase = new KieHelper().addContent(drl, ResourceType.DRL).build();
+        KieSessionConfiguration conf = KieServices.get().newKieSessionConfiguration();
+        conf.setOption(option);
+        session = kbase.newKieSession(conf, null);
+        savedSessionId = session.getIdentifier();
```
Comment (tkobayas) [inline drools-reliability/src/test/java/org/drools/reliability/ReliabilityTestBasics.java]: @nprentza Similar to my PR https://github.com/kiegroup/drools/pull/5104 , please let `savedSessionId` change only in `createSession`.

Comment (nprentza) [reply drools-reliability/src/test/java/org/drools/reliability/ReliabilityTestBasics.java]: @tkobayas I have applied the changes of PR #5014 into ReliabilityTestBasics. I will issue now a new PR.

**Pull request closed without merging**